### ISSUE TITLE
Explore private side capabilities of WordPress

### ISSUE THREAD

**Issue by RLC-DCPPC** [nih-cfde/public-website-content#17]
For the CFDE website, we want to include a private (in the authorization/authentication sense) side.

In order to determine if WordPress can meet our needs we need to: review the BioDataCatalyst website and determine, for each aspect, whether that capability can be accomplished with WordPress. Make sure to include evaluation of:

- [ ] shared calendar integration
- [ ] use of Globus Auth for authentication (see: [https://globus-integration-examples.readthedocs.io/en/latest/wordpress.html](https://globus-integration-examples.readthedocs.io/en/latest/wordpress.html))
- [ ] integration with GitHub; can WordPress talk to GitHub?

**Comment by kloydrti** [comment]
Each of those 3 things are possible with WordPress. The current WordPress site is already connected to GitHub to get the latest version of the content.

**Comment by kloydrti** [comment]
There are kanban board plugins for wordpress, though all of the features might have an extra cost attached. I'm confused about why we'd be replicating that kind of feature in a private-side wordpress site when it's already available directly on github. Why not create a github project and have people use that?

**Comment by ACharbonneau** [comment]
different levels of access

**Comment by RLC-DCPPC** [comment]
On the topic of the kanban plugins, is it a bi-directional integration, in the sense that the two boards are synchronized and a change made in GH would be replicated in WP and vice-versa?

**Comment by RLC-DCPPC** [comment]
Really want the private side to be an easy way to get to our stuff (in particular, GH). Some kind of address book person lookup, on-boarding site stuff, resources that we use all the time. Maybe once a month (deliverable status when done, e.g.). Weekly updates of %-complete using two-way interface.

DCCs will have access to CFDE Slack. But the easy interface could have a pointer to the CFDE Everyone Google Drive. E.G. RFC process; would like feedback from many DCCs (including unfunded ones). Private layer access to all DCCs - look at GH, see and edit Google docs. Funded DCCs would have read/write access to GH, Google. Permission worked out by Amanda.

**Comment by RLC-DCPPC** [comment]
BioDataCatalyst DOES have the two-way interface with GH

**Comment by RLC-DCPPC** [comment]
[Latest requirements.](https://docs.google.com/document/d/1rmJvgRcNVj2_PP8el26BbQiIM9EQH3AAkJ3D38aMqBU/edit)

**Comment by RLC-DCPPC** [comment]
7/7/20

- [ ] - Katie will explore Globus for Auth (maybe use Google account as username for Globus?)
- [ ] - Katie will start implementing the straightforward aspects.

**Comment by kloydrti** [comment]
Some notes about Globus -
I was able to create a Globus ID with my existing Google account. The Globus login screen has a "Sign in with Google" button that will create a new Globus account with your Google email. 

There is also an option to create a GlobusID account without linking Google or any other organization, but this creates a separate account that you would have to remember log in information for. 

I recommend that since everyone who is onboarded to CFDE is asked to set up Google anyway, we should instruct people to sign into Globus using that Google account. That way it will all stay connected and hopefully simple. There is an option to link multiple accounts to one Globus account, so someone could initially sign in with google and later add their organization-specific accounts.

I was able to use this process to log in to the Wordpress site. I can share my screen and run through it all in our website meeting tomorrow.

**Comment by RLC-DCPPC** [comment]
Hi @rpwagner , I'd like you to be aware of the the preceding comment from Katie, since you are the person on the team I most associate with Globus, in case you have any relevant knowledge to share, but mainly FYI.

**Comment by rpwagner** [comment]
(Adding @lliming)
Thanks for pointing this out. @kloydrti, Globus has the concept of account linking, so that if users want to create a GlobusID, they can, although that's not necessary in most cases. And if they do, they can still log in with one of their other IDs, like the Google one you mention. Although the preferred mechanism is for users to create their Globus accounts using one of their institutional credentials. For example, both NIH and eRA Commons identities are supported, along with most research universities. There's a lot more complexity to this we can go into.

**Comment by RLC-DCPPC** [comment]
7/14/20

Login through Google
Groups.io calendar in read-only mode

**Comment by RLC-DCPPC** [comment]
8/11/20

New reqs in purple [here](https://docs.google.com/document/d/1IYuN5C3YYnhMciaPCrpf-0S6cDeHr5Q1JYGtmjAdUyo/edit#bookmark=id.tzcrvt7rw6cb).

**Comment by RLC-DCPPC** [comment]
do we want to approve comments or let them fly.

**Comment by ACharbonneau** [comment]
do you mean in the RFCs?

**Comment by kloydrti** [comment]
Yes, the RFCs will have comments allowed. They can all be automatically approved, automatically approved for users with previously approved comments, or all held for review by a site manager regardless of author.

**Comment by ACharbonneau** [comment]
The comments should all appear as comments on the google doc. They should only be closed/accepted/incorporated into the doc by a human.

**Comment by RLC-DCPPC** [comment]
Planning for a demo this week 9/10 at 12:30.

**Comment by RLC-DCPPC** [comment]
9/10:  Demo notes (@ACharbonneau, @owhite, @amahurkar , @kloydrti , please review and repair as you will, Italics indicate "things to do" :-))

- "accepted" (1st class) RFCs are available to the hoi-poloi; logged in users can see more (open) of them
- _"planning statements" (like RFC's) - approved to public; add working versions to private side_
-- _put under project management_
- accounts with project manager permissions
- _can we make it so events can be public or private?
-- yes you can publish info but no booking feature (need a login to book)_
- onboarding form is scraped from Google form; available only to designated mangers
-_- need instructions here_
- sends submissions to Google sheet (cool!)
- fields from oboarding form will auto-fill into user's profile
-- can make some of these fields non-editable
-- _look into notify on change (update Google sheet, e.g.)_
--- can notify on Google sheet update (AC)
- can click affiliation/position and see others with same metafield value
- can view team members for PIs
- _what is our plan
-- we think that the DCCs are going to start having money in 3 weeks or so. 
-- So, what would be cool is if we could onboard through this.
-- welcome to CFDE website - we could move it over here_
_-- calendar, too_
_-- reg and landing pages for x-p events_
_-- keep tabs with Victor on re-hosting to AWS_
_-- if we decided that this is workable; move the site and make these two things
-- would like to also work on DCC project management page 
--- for now, a list of links, since we don;t see a good ZH API to build off of_

**Comment by ACharbonneau** [comment]
I've thought about it some more, and without a proper plugin, I think we should bail on the github integration for now...links will only work for people who are fully onboarded, so maybe we will have a github/zenhub link somewhere in the PM section, but I don't think it deserves its own page or any additional time at the moment

**Comment by RLC-DCPPC** [comment]
9/22

So, we have the Wordpress rehosted to AWS in dev. www-dev.nih-cfde.org.

@victor73, can we get Wordpress in staging environment soon, so Katie can get back to updating in dev? I suppose prod can wait a little longer.  Do we already know what the landing page URL will be so Katie can plan what to link to from the public (AKA brochure, AKA marketing) site?

**Comment by ACharbonneau** [comment]
Can you tag me with the new URL as well? I'll need to update it on the training/usecase/etc sites
